test/common_tests.dart

```diff
@@ -2188,7 +2188,7 @@ void runCommonTests(
             isSinkClosed = true;
           });
 
-          test('throwsIfEncodingIsNullAndWriteObject', () {
+          test('throwsIfEncodingIsNullAndWriteObject', () async {
             sink.encoding = null;
             expect(() => sink.write('Hello world'), throwsNoSuchMethodError);
```

Comment:
This is the error message 
`  This test failed after it had already completed. Make sure to use [expectAsync]
  or the [completes] matcher when testing async code.`

while `sink.write` does not return a future, the [docs](https://api.dartlang.org/stable/1.23.0/dart-io/IOSink/write.html) say it is non-blocking and have seen code where `sink.done` is called after a `write` to check if the write was done. Could not use `done` here due to the nature of the check, and it appeared that making it async forced the test framework to wait for completion. There have been optimizations in the 1.24 sdk with async that have caused such test failures, where code is async but not waited on to finish.